infrastructure/haproxy_reload.sh

```diff
@@ -1 +1,79 @@
-eval ${HAPROXY_RELOAD_COMMAND}
+#!/bin/bash
+
+num_rules=2
+real=0
+stats=1
+
+a_prefix=855
+b_prefix=866
+
+iptables_status() {
+   a=$(iptables -t nat -L -n -v | grep -c REDIRECT.*${a_prefix})
+   b=$(iptables -t nat -L -n -v | grep -c REDIRECT.*${b_prefix})
+
+   if   [[ ${a} == 0 && ${b} == 0 ]]; then echo "none"
+   elif [[ ${a} == ${num_rules} && ${b} == ${num_rules} ]]; then echo "both"
+   elif [[ ${a} == ${num_rules} ]]; then echo "a"
+   elif [[ ${b} == ${num_rules} ]]; then echo "b"
+   else echo "unknown"
+   fi
+}
+
+add() {
+  instance=$1
+  instance_prefix="${instance}_prefix"
+  real_port="${!instance_prefix}${real}"
+  stats_port="${!instance_prefix}${stats}"
+  iptables -t nat -A PREROUTING -m state --state NEW -p tcp -d ${HOST_IP} --dport 80 -j REDIRECT --to ${real_port}
+  iptables -t nat -A PREROUTING -m state --state NEW -p tcp -d ${HOST_IP} --dport 81 -j REDIRECT --to ${stats_port}
+}
+
+remove() {
+  instance=$1
+  instance_prefix="${instance}_prefix"
+  real_port="${!instance_prefix}${real}"
+  stats_port="${!instance_prefix}${stats}"
+  iptables -t nat -D PREROUTING -m state --state NEW -p tcp -d ${HOST_IP} --dport 80 -j REDIRECT --to ${real_port}
+  iptables -t nat -D PREROUTING -m state --state NEW -p tcp -d ${HOST_IP} --dport 81 -j REDIRECT --to ${stats_port}
+}
+
```

Comment:
The only difference between the impls of "add()" and "remove()" is the flag "-A" vs. "-D", maybe that could be unified passing in that flag as argument?